alacritty/src/window.rs

```diff
@@ -254,22 +262,28 @@ impl Window {
 
         let class = &window_config.class;
 
-        let mut builder = WindowBuilder::new()
+        let builder = WindowBuilder::new()
             .with_title(title)
             .with_visible(false)
             .with_transparent(true)
             .with_decorations(window_config.decorations != Decorations::None)
             .with_maximized(window_config.maximized())
             .with_fullscreen(window_config.fullscreen())
-            .with_window_icon(icon.ok())
-            // X11.
-            .with_class(class.instance.clone(), class.general.clone())
-            // Wayland.
-            .with_app_id(class.instance.clone());
```

Comment:
Honestly I feel like this is almost a mistake in winit to expose these things differently like that. Would it maybe make sense to think about unifying these upstream to have a `with_class` that sets `app_id` on Wayland?

Reply:
Maybe it can accept some kind of enum, and have them in one call, but the main issue is that those things have different amount of arguments and don't really relate to each other.

Reply:
How do they not relate to each other? Isn't `app_id` basically the Wayland class equivalent? And I think we can resolve the different parameter issue with thorough documentation.

As you can see we need to pass both anyways since we want to work on X11 and winit isn't a toolkit that should be used to build applications that work just one Wayland. So there should be abstractions and I think this abstraction would make sense.

Is there any scenario where you would want to set the class instance to something different than the `app_id`?

Reply:
I guess you can call it as alternative, I'm just saying that there's no instance thing. You can propose unification upstream.

Reply:
I will, thanks.